### PR TITLE
[CORE-664] Add comment/remove unnecessary lines in `x/perpetual`

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
- Add additional comments for readability, broken invariant
- Removed staled TODO items
- Removed unused variables
- Added a panic where an invariant is broken

### Test Plan
N/A

### Author/Reviewer Checklist
- [x] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [x] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [x] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [x] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [x] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/x/perpetuals/keeper/msg_server_add_premium_votes.go [hunk 1]
```diff
@@ -9,7 +9,6 @@ import (
 )
 
 // AddPremiumVotes add new premium sample votes from a proposer to the application.
-// TODO(DEC-1310): Rename this message handler.
```
Comment (teddyding) [inline protocol/x/perpetuals/keeper/msg_server_add_premium_votes.go:12]: Name already reflects updated design.

File: protocol/x/perpetuals/keeper/perpetual.go [hunk 2]
```diff
@@ -293,7 +293,6 @@ func (k Keeper) processPremiumVotesIntoSamples(
 					metrics.PerpetualId,
 					int(perp.GetId()),
 				),
-				// TODO(DEC-1071): Add epoch number as label.
```
Comment (teddyding) [inline protocol/x/perpetuals/keeper/perpetual.go:296]: We decided not to add epoch number or anything that has unbonded cardinality as labels.

File: protocol/x/perpetuals/types/constants.go [hunk 3]
```diff
@@ -11,10 +9,3 @@ const (
 	// TODO(DEC-1105): Move this constant to state so that it can be changed via governance.
 	RemovedTailSampleRatioPpm uint32 = 0
 )
-
-// TempBigImpactNotionalAmount is a temporary function that returns 5_000 USDC as
-// the impact notional amount.
-// TODO(DEC-1308): Remove this.
-func TempBigImpactNotionalAmount() *big.Int {
```
Comment (teddyding) [inline protocol/x/perpetuals/types/constants.go:18]: Not used anymore

File: protocol/x/perpetuals/types/liquidity_tier.go [hunk 4]
```diff
@@ -52,6 +54,13 @@ func (liquidityTier LiquidityTier) GetMaintenanceMarginPpm() uint32 {
 // `|S| ≤ Clamp Factor * (Initial Margin - Maintenance Margin)`, which can be applied to both
 // funding rate clamping and premium vote clamping, each having their own clamp factor.
 func (liquidityTier LiquidityTier) GetMaxAbsFundingClampPpm(clampFactorPpm uint32) *big.Int {
+	maintenanceMarginPpm := liquidityTier.GetMaintenanceMarginPpm()
+	if maintenanceMarginPpm > liquidityTier.InitialMarginPpm {
```
Comment (teddyding) [inline protocol/x/perpetuals/types/liquidity_tier.go:58]: Invariant check such that line 67 does not overflow

**Review by tqin7 — APPROVED** [review]

**Pull request merged**